lib/modules/manager/npm/extract/pnpm.ts

```diff
@@ -135,3 +142,124 @@ export async function detectPnpmWorkspaces(
     }
   }
 }
+
+export async function getPnpmLock(filePath: string): Promise<LockFile> {
+  // TODO #7154
+  const pnpmLockRaw = (await readLocalFile(filePath, 'utf8'))!;
+  try {
+    const lockParsed = load(pnpmLockRaw);
+    if (!isPnpmLockfile(lockParsed)) {
+      throw new Error('Invalid or empty lockfile');
+    }
+    logger.trace({ lockParsed }, 'pnpm lockfile parsed');
+
+    let lockfileVersion = lockParsed.lockfileVersion;
+    // field lockfileVersion is type string in lockfileVersion 6 and type number in <6
```

Comment:
why can't we just leave as string?

Reply:
lockfileVersion<6 store the value as number. After parsing using the `load` fn we get a number for version<6 and string for versino=6.